isis/src/mro/apps/hical/hical.cpp

```diff
@@ -0,0 +1,530 @@
+/** This is free and unencumbered software released into the public domain.
+
+The authors of ISIS do not claim copyright on the contents of this file.
+For more details about the LICENSE terms and the AUTHORS, you will
+find files of those names at the top level of this repository. **/
+
+/* SPDX-License-Identifier: CC0-1.0 */
+
+#include <cstdio>
+#include <QString>
+#include <vector>
+#include <algorithm>
+#include <sstream>
+#include <iostream>
+
+#include "Application.h"
+#include "FileName.h"
+#include "ProcessByLine.h"
+#include "UserInterface.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "IString.h"
+#include "HiCalConf.h"
+#include "CollectorMap.h"
+#include "HiCalTypes.h"
+#include "HiCalUtil.h"
+#include "HiCalData.h"
+#include "Statistics.h"
+#include "SplineFill.h"              // SpineFillComp.h
+#include "LowPassFilter.h"           // LowPassFilterComp.h
+#include "ZeroBufferSmooth.h"        // DriftBuffer.h  (Zf)
+#include "ZeroBufferFit.h"           // DriftCorrect.h (Zb)
+#include "ZeroReverse.h"             // OffsetCorrect.h (Zz)
+#include "ZeroDark.h"                // DarkSubtractComp.h (Zd)
+#include "ZeroDarkRate.h"            // DarkSubtractComp.h (Zdr)
+#include "GainLineDrift.h"           // GainVLineComp.h (Zg)
+#include "GainNonLinearity.h"        // Non-linear gain (new)
+#include "GainChannelNormalize.h"    // ZggModule.h (Zgg)
+#include "GainFlatField.h"           // FlatFieldComp.h (Za)
+#include "GainTemperature.h"         // TempGainCorrect.h (Zt)
+#include "GainUnitConversion.h"      // ZiofModule.h (Ziof)
+
+#include "hical.h"
+
+using namespace std;
+
+namespace Isis {
+
+  //!< Define the matrix container for systematic processing
+  typedef CollectorMap<IString, HiVector, NoCaseStringCompare> MatrixList;
+
+  void hical(UserInterface &ui, Pvl *log) {
+    const QString hical_program = "hical";
+    const QString hical_version = "5.0";
+    const QString hical_revision = "$Revision: 6715 $";
+    const QString hical_runtime = Application::DateTime();
+
+    QString procStep("prepping phase");
+    MatrixList *calVars = nullptr;
+    try {
+//  The output from the last processing is the input into subsequent processing
+      ProcessByLine p;
+
+      Cube *hifrom = p.SetInputCube(ui.GetFileName("FROM"), ui.GetInputAttribute("FROM"));
+      int nsamps = hifrom->sampleCount();
+      int nlines = hifrom->lineCount();
+
+//  Initialize the configuration file
+      QString conf(ui.GetAsString("CONF"));
+      HiCalConf hiconf(*(hifrom->label()), conf);
+      DbProfile hiprof = hiconf.getMatrixProfile();
+
+// Check for label propagation and set the output cube
```

Comment:
Not to be petty, but it might be worth adjusting all these comments to line up with the code. Seems like most of them are not tabbed over